src/datepicker/datepicker.ts

```diff
@@ -222,8 +260,21 @@ export class NgbDatepicker implements OnChanges,
     }
   }
 
-  private _updateData() {
-    this.month = this._service.generateMonthViewModel(
-        this._date, this._minDate, this._maxDate, toInteger(this.firstDayOfWeek), this.markDisabled);
+  private _updateData(force = false) {
+    const newMonths = [];
+    for (let i = 0; i < this.displayMonths; i++) {
+      const newDate = this._calendar.getNext(this._date, 'm', i);
+      const index = this.months.findIndex(month => month.firstDate.equals(newDate));
+
+      if (force || index === -1) {
+        newMonths.push(
+            this._service.generateMonthViewModel(
+                newDate, this._minDate, this._maxDate, toInteger(this.firstDayOfWeek), this.markDisabled));
+      } else {
+        newMonths.push(this.months[index]);
+      }
+    }
+
+    this.months = newMonths;
```

Comment:
Hmm, me neither. Not sure if it hurts readability or not. It looks nice for real array-like objects and as a shortcut for simple initialization as in rating.